src/main/scala/com/mozilla/telemetry/streaming/CrashesToInflux.scala

```diff
@@ -0,0 +1,188 @@
+      descr = "Max number of parallel requests in batch mode",
+      required = false,
+      default = Some(100))
+
+    conflicts(kafkaBroker, List(from, to, fileLimit, minDelay, maxParallelRequests))
+    validateOpt (sample) {
+      case Some(s) if 0.0 < s && s <= 1.0 => Right(Unit)
+      case Some(o) => Left(s"Sample out of range. Expected 0.0 < Sample <= 1.0, Found $o")
+    }
+
+    verify()
+  }
+
+  def parsePing(message: Message): Array[String] = {
+    val fields = message.fieldsAsMap
+
+    if (!fields.get("docType").contains("crash")) {
+      throw new Exception("Ping must be crash type")
+    }
+
+    val ping = CrashPing(message)
+    val meta = ping.meta
+
+    //if (meta.normalizedChannel != "beta") { // TODO: only look at beta for now
+    //  return Array[String]()
+    //}
+
+    val timestamp = meta.Timestamp
+
+    val influxFields = mutable.LinkedHashMap(
+      "buildId" -> ping.getNormalizedBuildId.getOrElse("")
+    )
+
+    val influxTags = mutable.LinkedHashMap(
+      "submissionDate" -> meta.submissionDate,
+      "appVersion" -> meta.appVersion,
+      "displayVersion" -> ping.getDisplayVersion.getOrElse(""),
+      "channel" -> meta.normalizedChannel,
+      "country" -> meta.geoCountry,
+      "osName" -> meta.os.getOrElse(""),
+      "osVersion" -> ping.getOsVersion.getOrElse(""),
+      "architecture" -> ping.getArchitecture.getOrElse("")
+    )
+
+    val outputString = new StringBuilder("crashes") // TODO: make measurement name configurable
+
+    influxTags.foreach({
+      case (k, v) => outputString.append(s",$k=$v")
+    })
+    influxFields.zipWithIndex.foreach({
+      case ((k, v), i) =>
+        val prefix = if (i == 0) " " else ","
+        outputString.append(s"$prefix$k=$v")
+    })
+    outputString.append(s" $timestamp")
+
+    Array[String](outputString.toString())
+  }
+
+  def getParsedPings(pings: DataFrame, raiseOnError: Boolean): Dataset[String] = {
+    import pings.sparkSession.implicits._
+
+    pings.flatMap( v => {
+      try {
+        parsePing(Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]]))
+      } catch {
+        case _: Throwable if !raiseOnError => Array[String]()
+      }
+    }).as[String]
+  }
+
+  def sendStreamCrashes(spark: SparkSession, opts: Opts): Unit = {
+    val httpSink = new HttpSink(opts.url(), Map())
+
+    val pings = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", opts.kafkaBroker())
+      .option("failOnDataLoss", opts.failOnDataLoss())
+      .option("kafka.max.partition.fetch.bytes", 8 * 1024 * 1024) // 8MB
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", TelemetryKafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    getParsedPings(pings.select("value"), opts.raiseOnError())
+      .writeStream
+      .queryName(QueryName)
+      .foreach(httpSink)
```

Comment:
As you mentioned, batching these events might be a good idea.
Unfortunetely we can't use foreachPartition on streaming dataset, so this would have to be done in the HttpSink, if needed.